src/lints/trait_method_missing.ron

```diff
@@ -0,0 +1,58 @@
+SemverQuery(
+    id: "trait_method_missing",
+    human_readable_name: "pub trait method removed or renamed",
+    description: "A trait method is no longer visible, or can no longer be accessed by its prior path.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#major-any-change-to-trait-item-signatures"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        method {
+                            method_visibility: visibility_limit @filter(op: "=", value: ["$public"]) @output
+                            method_name: name @output @tag
+
+                            span_: span @optional {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
+                    }
+                }
+            }
+            current @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
```

Comment:
Recall that we're aiming for only a single lint to be triggered for any given semver violation.

With this `@fold @transform(op: "count") @filter(op: "=", value: ["$zero"])` here, I believe it's possible to trigger this lint in a case that _doesn't involve_ removing or renaming a trait method.

Can you think of any such example situations? They'd be great to include in the test suite even after updating the query here.

Reply:
ack